client/site-profiler/hooks/use-site-profiler-record-analytics.ts

```diff
@@ -0,0 +1,26 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useEffect } from 'react';
+import type { CONVERSION_ACTION } from 'calypso/site-profiler/hooks/use-define-conversion-action';
+
+/**
+ * Record analytics events for site-profiler.tsx root element only
+ * The idea is to move all root element analytics events to this hook
+ */
+export default function useSiteProfilerRecordAnalytics(
+	domain: string,
+	isDomainValid?: boolean,
+	conversionAction?: CONVERSION_ACTION
+) {
+	useEffect( () => {
+		recordTracksEvent( 'calypso_site_profiler_page_view' );
+	}, [] );
+
+	useEffect( () => {
+		domain &&
+			isDomainValid &&
+			recordTracksEvent( 'calypso_site_profiler_domain_analyze', {
+				domain,
+				conversion_action: conversionAction,
```

Comment:
Since there's not only `conversionAction` but also `specialDomainMapping` now, should we handle this as separate event_props or adding another one?

Reply:
I'm getting wrong `conversion_action` recorded here when it triggered, I think it is because it triggered too early? Because the final  `conversion_action` depends on multiple API results like WHOIS data, hosting provider and urlData?

Reply:
I've improved the logic of defining conversion action. Now it should work properly!